v0/destinations/custify/util.js

```diff
@@ -0,0 +1,221 @@
+const get = require("get-value");
+const {
+  ConfigCategory,
+  MappingConfig,
+  ReservedTraitsProperties,
+  ReservedCompanyProperties
+} = require("./config");
+const { httpPOST } = require("../../../adapters/network");
+const {
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+const {
+  CustomError,
+  getFieldValueFromMessage,
+  removeUndefinedAndNullValues,
+  constructPayload,
+  isHttpStatusSuccess
+} = require("../../util");
+
+/**
+ *
+ * Craete or update company based on the group call parameters
+ * @param {*} companyPayload
+ * @param {*} Config
+ * @api https://docs.custify.com/#tag/Company/paths/~1company/post
+ */
+const createUpdateCompany = async (companyPayload, Config) => {
+  const companyResponse = await httpPOST(
+    ConfigCategory.GROUP_COMPANY.endpoint,
+    companyPayload,
+    {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${Config.apiKey}`
+      }
+    }
+  );
+  const processedCompanyResponse = processAxiosResponse(companyResponse);
+  if (!isHttpStatusSuccess(processedCompanyResponse.status)) {
+    const errMessage = JSON.stringify(processedCompanyResponse.response) || "";
+    const errorStatus = processedCompanyResponse.status || 500;
+    throw new CustomError(
+      `[Group]: failed create/update company details ${errMessage}`,
+      errorStatus
+    );
+  }
+};
+
+const getCompanyAttribute = (companyId, remove = false) => {
+  if (companyId) {
+    const companiesList = [];
+    companiesList.push({
+      company_id: companyId,
+      remove
+    });
+    return companiesList;
+  }
+  return null;
+};
+
+/**
+ * This function is used for post processing the user payload
+ * the role of this fucntion is to format name and company id
+ * also it sets the custom attributes to the user payload
+ * @param {*} userPayload
+ * @param {*} message
+ * @returns {*} userPayload
+ */
+const postPorcessUserPayload = (userPayload, message) => {
+  const finalPayload = userPayload;
+
+  if (!userPayload.user_id && !userPayload.email) {
+    throw new CustomError("Email or userId is mandatory", 400);
+  }
+  if (userPayload.name === undefined || userPayload.name === "") {
+    const firstName = getFieldValueFromMessage(message, "firstName");
+    const lastName = getFieldValueFromMessage(message, "lastName");
+    if (firstName && lastName) {
+      finalPayload.name = `${firstName} ${lastName}`;
+    } else {
+      finalPayload.name = firstName || lastName;
+    }
```

Comment:
What if both are missing?

Reply:
it will be set as undefined and get cleared at the end